server/src/data-source.ts

```diff
@@ -0,0 +1,17 @@
+import "reflect-metadata"
+import { DataSource } from "typeorm"
+import { User } from "./entity/User"
+
+export const AppDataSource = new DataSource({
+    type: "postgres",
+    host: "localhost",
+    port: 5432,
+    username: process.env.DB_USERNAME || 'root',
+    password: process.env.DB_PASSWORD || 'root',
+    database: process.env.DB_DATABASE || 'shop',
```

Comment:
It's not that bad, but I would recommend putting it in constants above `AppDataSource` like that:
```suggestion

const DB_USERNAME = process.env.DB_USERNAME || 'root'
const DB_PASSWORD = process.env.DB_PASSWORD || 'root'
const DB_DATABASE = process.env.DB_DATABASE || 'shop'

export const AppDataSource = new DataSource({
    type: "postgres",
    host: "localhost",
    port: 5432,
    username: DB_USERNAME,
    password: DB_PASSWORD,
    database: DB_DATABASE,
```